src/hooks/test/use-click-away-test.js

```diff
@@ -62,7 +60,23 @@ describe('useClickAway', () => {
       });
       wrapper.update();
 
-      assert.equal(handler.callCount, 0);
+      assert.notCalled(handler);
+    });
+
+    it(`should not invoke callback on elements removed from container (${event.type})`, () => {
+      const wrapper = createComponent();
+      const button = wrapper.find('button').getDOMNode();
+
+      act(() => {
+        button.addEventListener(event.type, () => {
+          // Remove the button from the DOM before next listeners are invoked
```

Comment:
I would suggest to note this simulates what happened if the clicked element was removed as a result of a state update + re-render in a child component.